### PR TITLE
AE2 wires

### PR DESCRIPTION
**Pull request by ramidzkh**
AE2 wires which can carry channels while fitting into a pipe block

### CODE REVIEW

File: src/main/java/aztech/modern_industrialization/pipes/api/ExtraPipesInitializer.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package aztech.modern_industrialization.pipes.api;
+
+public interface ExtraPipesInitializer {
```
Comment (Technici4n) [inline src/main/java/aztech/modern_industrialization/pipes/api/ExtraPipesInitializer.java:26]: I don't like this, it's cursed.

File: src/generated/resources/assets/modern_industrialization/lang/en_us.json [hunk 2]
```diff
@@ -1418,6 +1418,7 @@
   "tag.modern_industrialization.fluid_pipes": "Fluid Pipes",
   "tag.modern_industrialization.forge_hammer_tools": "Forge Hammer Tools",
   "tag.modern_industrialization.item_pipes": "Item Pipes",
+  "tag.modern_industrialization.me_wires": "ME Wire",
```
Comment (Technici4n) [inline src/generated/resources/assets/modern_industrialization/lang/en_us.json:1421]: Should `ME Wires` (plural)

File: src/generated/resources/data/modern_industrialization/recipes/compat/ae2/dyes/black/craft/me_wire_1.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+  "type": "minecraft:crafting_shaped",
+  "fabric:load_conditions": [
```
Comment (Technici4n) [inline src/generated/resources/data/modern_industrialization/recipes/compat/ae2/dyes/black/craft/me_wire_1.json:3]: Sadly this is not enough if the compat is disabled... But good enough until we can use FabricMC/fabric#2548.

File: src/main/java/aztech/modern_industrialization/compat/ae2/pipe/package-info.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+/**
+ * Implements a star topology between all pipes with any {@link aztech.modern_industrialization.pipes.api.PipeEndpointType#BLOCK} connection to a pipe
+ * network's own node
+ */
+package aztech.modern_industrialization.compat.ae2.pipe;
```
Comment (Technici4n) [inline src/main/java/aztech/modern_industrialization/compat/ae2/pipe/package-info.java:5]: We can move this somewhere else ^^

**Comment by ramidzkh** [comment]
Missing a few things right now: 
* Wires connecting to each other by default
* Wires recipes

Nice to have:
* Better models when connecting to AE2 cables
* Better wire textures

File: src/main/java/aztech/modern_industrialization/compat/ae2/pipe/MENetworkNode.java [hunk 5]
```diff
@@ -81,7 +81,7 @@ public void updateConnections(Level world, BlockPos pos) {
                     return true;
                 }
             }
-            return false;
+            return !canConnect(world, pos, connection);
```
Comment (Technici4n) [inline src/main/java/aztech/modern_industrialization/compat/ae2/pipe/MENetworkNode.java:84]: Intended feature

**Pull request merged**